examples/web_apps/webxr_mobile_demo/js/vr_demo.js

```diff
@@ -0,0 +1,423 @@
+    let rayHitInfo = this.raycastFromHead();
+    const hitObjId = rayHitInfo == null ? -1 : rayHitInfo.objectId;
+    if (hitObjId != -1) {
+      this.heldObjectId = hitObjId;
+      if (this.headRot != null) {
+        let currRot = this.sim.getRotation(this.heldObjectId, 0);
+        this.heldRelRot = Module.Quaternion.mul(
+          this.headRot.inverted(),
+          currRot
+        );
+      }
+      this.sim.setObjectMotionType(
+        Module.MotionType.KINEMATIC,
+        this.heldObjectId,
+        0
+      );
+    }
+  }
+
+  // When the user picks up their finger, we drop the held object if it exists.
+  onRelease() {
+    if (this.heldObjectId == -1) {
+      return;
+    }
+    this.sim.setObjectMotionType(
+      Module.MotionType.DYNAMIC,
+      this.heldObjectId,
+      0
+    );
+    this.simHelpers.hide(this.dropMarker);
+    this.heldObjectId = -1;
+  }
+
+  // Given the pose (position/orientation of the phone), update the agent's eye
+  // sensor location.
+  updatePose(pose, agent) {
+    const pointToArray = p => [p.x, p.y, p.z, p.w];
+    const FWD = Module.Vector3.zAxis(1);
+    const FWD_ANGLE = Math.atan2(FWD.z(), FWD.x());
+    const DOWN = Module.Vector3.yAxis(-1);
+    const viewYawOffset = 0;
+
+    const headRotation = Module.toQuaternion(
+      pointToArray(pose.transform.orientation)
+    );
+    const pointingVec = headRotation.transformVector(FWD);
+    const pointingAngle =
+      Math.atan2(pointingVec.z(), pointingVec.x()) - FWD_ANGLE;
+
+    const agentQuat = Module.Quaternion.rotation(
+      new Module.Rad(pointingAngle + viewYawOffset),
+      DOWN
+    );
+    const inverseAgentRot = Module.Quaternion.rotation(
+      new Module.Rad(-pointingAngle),
+      DOWN
+    );
+
+    let state = new Module.AgentState();
+    agent.getState(state);
+    state.rotation = Module.toVec4f(agentQuat);
+    agent.setState(state, false);
+
+    const view = pose.views[0];
+
+    const sensor = agent.getSubtreeSensors().get(VIEW_SENSOR);
+
+    let pos = pointToArray(view.transform.position).slice(0, -1); // don't need w for position
+    // adjust sensitivity
+    pos[0] *= 3; // x
+    pos[1] *= 1.5; // y
+    pos[2] *= 3; // z
+    sensor.setLocalTransform(
+      Module.toVec3f(
+        inverseAgentRot.transformVector(new Module.Vector3(...pos))
+      ),
+      Module.toVec4f(
+        Module.Quaternion.mul(
+          inverseAgentRot,
+          Module.toQuaternion(pointToArray(view.transform.orientation))
+        )
+      )
+    );
+
+    this.lookDir = new Module.Vector3(
+      -pointingVec.x(),
+      -pointingVec.y(),
+      -pointingVec.z()
+    );
+    this.headPos = Module.Vector3.add(
+      this.agentPos,
+      new Module.Vector3(...pos)
+    );
+    this.headRot = headRotation;
+  }
+
+  // Adjust the held object to stay in front of the cursor with the correct
+  // rotation.
+  updateHeldObjectPose() {
+    if (this.heldObjectId == -1) {
+      return;
+    }
+    function scale(vec, amount) {
```

Comment:
We should instead implement scalar multiplication in bindings_js.cpp, similar to the Vector3 `add` already there.